armi/reactor/converters/uniformMesh.py

```diff
@@ -131,11 +131,12 @@ def convert(self, r=None):
                 f"the core's reference assembly mesh: {r.core.refAssem.getAxialMesh()}"
             )
             self.convReactor = self._sourceReactor
+            self.convReactor.core.updateAxialMesh()
             self._setParamsToUpdate()
             for assem in self.convReactor.core.getAssemblies(self._nonUniformMeshFlags):
                 homogAssem = self.makeAssemWithUniformMesh(
                     assem,
-                    self.convReactor.core.refAssem.getAxialMesh(),
```

Comment:
Honest question:

@mgjarrett If this is all about submeshes, shouldn't this change of code (`getAxialMesh()` -> `p.axialMesh`) be wrapped in a check for submeshes?

(Something like `if block.p.axMesh > 1`)

Sorry, I'm just trying to make sure I fully understand the change.  Like: Should it _always_ be the new way, or does it still have to be the old way sometimes?

Sorry to bother!

Reply:
Good question. I cannot think of a reason why it would need to be the old way. If there is no submesh, `p.axialMesh` and `getAxialMesh()` should be equivalent.

Since the assembly created by `makeAssemWithUniformMesh` never has a submesh, we should homogenize it onto the `p.axialMesh` (which also has no submesh), instead of the `getAxialMesh()`.